File: packages/og-injector-rust/src/server/models/image_data.rs
use serde::Deserialize;
use url::Url;
use url_serde;

#[derive(Deserialize)]
pub struct ImageData {
    pub height: u64,
    pub width: u64,
    pub mime_type: String,
    pub srcs: ImageSrcData,
}

#[derive(Deserialize)]
#[allow(dead_code)]
pub struct ImageSrcData {
    #[serde(with = "url_serde")]
    pub full: Url,
    #[serde(with = "url_serde")]
    pub high: Option<Url>,
    #[serde(with = "url_serde")]
    pub low: Option<Url>,
    #[serde(with = "url_serde")]
    pub gallery: Option<Url>,
}

File: packages/og-injector/src/models/ImageData.ts
export interface ImageData {
  id: string
  position: number
  filename: string
  height: number
  width: number
  mime_type: string
  ext: string

  srcs: ImageSizes
}

type ImageSizes =
  & { full: string }
  & { [size: string]: string | undefined }
